python/pybind11/cucim_py.cpp

```diff
@@ -395,6 +395,22 @@ py::object py_read_region(const CuImage& cuimg,
     }
 }
 
+py::object py_associated_image(const CuImage& cuimg, const std::string& name, const io::Device& device)
+{
+    auto image_ptr = std::make_shared<cucim::CuImage>(cuimg.associated_image(name, device));
+
+    {
+        py::gil_scoped_acquire scope_guard;
+
+        py::object image = py::cast(image_ptr);
+
+        // Add `__array_interace__` or `__cuda_array_interface__` in runtime.
+        _set_array_interface(image);
```

Comment:
Wondering if we should acquire the GIL within `_set_array_interface`. WDYT?

Reply:
The line 403 (https://github.com/rapidsai/cucim/pull/65/files#diff-48de51ebe61a16c8d3878cba6d31c628f024287fc2671440f9bae39ab6bcbd64R403 ) is for acquiring GIL :)

Reply:
Was thinking about this ( https://github.com/rapidsai/cucim/blob/372b39d4d196145f7c7cbc76b27c1abe9cb89183/python/pybind11/cucim_py.cpp#L104-L106 ), but I guess we already don't release the GIL (so should have it). So agree it's a non-issue